src/mpyl/artifacts/__init__.py

```diff
@@ -6,23 +6,26 @@
 from ..utilities.repo import Repository
 
 
+CACHE_FOLDER_NAME = "cache"
+
+
 class BuildArtifacts:
     logger: Logger
     codebase_repo: Repository
     artifact_repo: Repository
-    artifact_repo_path: Path
+    cache_folder: Path
 
     def __init__(
         self,
         logger: Logger,
         codebase_repo: Repository,
         artifact_repo: Repository,
         artifact_repo_path: Path,
-    ):  # codebase repo might not be needed
+    ):
         self.logger = logger
         self.codebase_repo = codebase_repo
         self.artifact_repo = artifact_repo
-        self.artifact_repo_path = artifact_repo_path
+        self.cache_folder = artifact_repo_path / CACHE_FOLDER_NAME
```

Comment:
Can't we use a temp folder for this?

Reply:
What do you mean precisely? I needed a subfolder inside the repo for handling all the build artifacts logic (like deleting everything before a copy) and avoid issues with the existing files that you get from cloning (like the readme and .gitignore).

Reply:
A temp folder is located outside the repo path https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory

Reply:
It's not a temp folder, how am I supposed to put the folder which contains the files that need to be pushed/pulled from/in outside the repo?

Reply:
create temp folder, clone or create repo in it, copy files, commit, push

Reply:
Ahh, I understood it differently because you commented on a folder that is _inside_ the repo folder. I'll take a look at this TemDir python thing, didn't know about it!